tools/lsp/preview/native.rs

```diff
@@ -117,284 +192,144 @@ pub fn load_preview(
-    let r = cache.source_code.get(&path).cloned();
-    cache.dependency.insert(path);
-    r
+    });
 }
 
-#[derive(Default)]
-struct PreviewState {
-    handle: Option<slint_interpreter::ComponentInstance>,
+fn get_current_style() -> String {
+    PREVIEW_STATE.with(|preview_state| -> String {
+        let preview_state = preview_state.borrow();
+        if let Some(ui) = &preview_state.ui {
+            ui.get_current_style().as_str().to_string()
+        } else {
+            String::new()
+        }
+    })
 }
-thread_local! {static PREVIEW_STATE: std::cell::RefCell<PreviewState> = Default::default();}
 
-pub fn design_mode() -> bool {
-    let cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
-    cache.design_mode
+pub fn set_busy(busy: bool) {
+    i_slint_core::api::invoke_from_event_loop(move || {
+        PREVIEW_STATE.with(|preview_state| {
+            let preview_state = preview_state.borrow_mut();
+            if let Some(ui) = &preview_state.ui {
+                ui.set_is_busy(busy);
+            }
+        });
+    })
+    .unwrap();
 }
 
-pub fn set_design_mode(sender: crate::ServerNotifier, enable: bool) {
-    let mut cache = CONTENT_CACHE.get_or_init(Default::default).lock().unwrap();
-    cache.design_mode = enable;
+pub fn send_status(message: &str, health: Health) {
+    let Some(sender) = SERVER_NOTIFIER.get_or_init(Default::default).lock().unwrap().clone() else {
+        return;
+    };
 
-    configure_design_mode(enable, &sender);
-    send_notification(
-        &sender,
-        if enable { "Design mode enabled." } else { "Design mode disabled." },
-        Health::Ok,
-    );
+    crate::preview::send_status_notification(&sender, message, health)
 }
 
-fn show_document_request_from_element_callback(
-    file: &str,
+pub fn ask_editor_to_show_document(
+    file: String,
     start_line: u32,
     start_column: u32,
-    _end_line: u32,
-    _end_column: u32,
-) -> Option<lsp_types::ShowDocumentParams> {
-    use lsp_types::{Position, Range, ShowDocumentParams, Url};
-
-    let start_pos = Position::new(start_line.saturating_sub(1), start_column.saturating_sub(1));
-    // let end_pos = Position::new(end_line.saturating_sub(1), end_column.saturating_sub(1));
-    // Place the cursor at the start of the range and do not mark up the entire range!
-    let selection = Some(Range::new(start_pos, start_pos));
-
-    Url::from_file_path(file).ok().map(|uri| ShowDocumentParams {
-        uri,
-        external: Some(false),
-        take_focus: Some(true),
-        selection,
-    })
+    end_line: u32,
+    end_column: u32,
+) {
+    let Some(sender) = SERVER_NOTIFIER.get_or_init(Default::default).lock().unwrap().clone() else {
+        return;
+    };
+
+    let fut = crate::send_show_document_to_editor(
+        sender,
+        file,
+        start_line,
+        start_column,
+        end_line,
+        end_column,
+    );
+
+    let _ = slint_interpreter::spawn_local(fut); // Fire and forget.
```

Comment:
the error here is an error that we couldn't spawn the future.   I think it shouldn't be ignored.
This only happen if we're not in the right thread or that there isn't any eventloop

```suggestion
    slint_interpreter::spawn_local(fut).unwrap();
```